keps/sig-network/20201014-NamespaceAsName.md

```diff
@@ -0,0 +1,348 @@
+```
+
+which is referenced from the namespaceSelector:
+
+```
+    + NamespaceSelector *NamespaceSelector
+    - NamespacesSelector *metav1.LabelSelector
+```
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As an administrator I want to block all users from accessing the kube-system namespace as a fundamental default policy.
+
+#### Story 2
+
+As an user I want to "just add" a namespace to my allow list without having to manage the labels which might get added/removed over time from said namespace.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+- CNIs may not choose, initially to support this construct, and dilligent communication with CNI providers will be needed to make sure its widely adopted.
+- Fractionating the Kubernetes API idiom by introducing a separate way to select objects.  We accept this cost because security is a fundamentally important paradigm that justifies breaking other paradigms, at times. 
+
+## Design Details
+
+- Add a new selector to the network policy peer data structure which can switch between allowing a "matchName" or "matchLabels" implementation, supporting a policy that is expressed like this:
+
+This selector has two, EXCLUSIVE, possible input types:
+
+- A list of conventional namespaces OR 
+- A `*` with an `exclude` rule as in input to this selector
+
+The "conventional namespaces" will look like so:
+```
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: mysql-allow-app
+spec:
+  podSelector:
+    matchLabels:
+      app: mysql
+  ingress:
+  - from:
+    - namespaceSelector:
+         matchName:
+         -  my-frontend
+```
+
+Meanwhile, the "exclude" implementation, will look like this:
+
+```
+  ingress:
+  - from:
+    - namespaceSelector:
+         matchName:
+         -  *
+         exclude:
+         - kube-system
+```
+
+
+### Test Plan
+
+We will add tests for this new api semantic into the exting test/e2e/ network policy test suites in upstream which cover both of these scenarios, using the validation framework outlined in https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/20200204-NetworkPolicy-verification-rearchitecture.md#motivation 
+
+### Graduation Criteria
+
+#### Alpha 
+- Add a feature gated new field to NetworkPolicy
+- Communicate CNI providers about the new field
+- Add validation tests in API
+
+#### Beta
+- The name selector has been supported for at least 1 minor release
+- At least one CNI provider implements the new field
+- Feature Gate is enabled by Default
+
+#### GA Graduation
+
+- At least two CNIs support the The name selector field
+- The name selector has been enabled by default for at least 1 minor release
+
+#### Removing a Deprecated Flag
```

Comment:
(this section is irrelevant to this KEP and can be removed)

Reply:
fixed